### PR TITLE
CI: added github-action to push geth-with-protocol image to dockerhub

### PR DESCRIPTION
**Pull request by RiccardoBiosas**
**Context**
It adds a github action to push the geth-with-protocol image to dockerhub

### CODE REVIEW

File: .github/workflows/geth-with-protocol.yml
Comment (leszko) [inline .github/workflows/geth-with-protocol.yml:5]: I think this will rebuild the image after every push to `master`.
- If we want to do it, then the image should be `livepeer/geth-with-livepeer-protocol:latest` (not `confluence`)
- If we don't want to do it, then I'd set the GH Action to trigger manually (you can also consider building on pushing a tag, but the issue here is that this repo contains a lot of other things, so I'm not sure we want to tag only for `livepeer/geth-with-livepeer-protocol`

**Review by leszko — COMMENTED** [review]
Added one comment. Other, then that, I wonder, how did you test this GH Action if it works?

File: .github/workflows/geth-with-protocol.yml
Comment (RiccardoBiosas) [reply .github/workflows/geth-with-protocol.yml:5]: good point, i think that the second option is the most sensible. also checking out the old [geth-with-livepeer](https://hub.docker.com/r/livepeer/geth-with-livepeer-protocol/tags) it doesn't seem it was following any particular tag system other than a new release for a breaking change

**Comment by RiccardoBiosas** [comment]
> Added one comment. Other, then that, I wonder, how did you test this GH Action if it works?

I only validated whether the YAML/syntax are correct. I'm not aware of any method to test a GH with shared secrets locally

**Comment by leszko** [comment]
> > Added one comment. Other, then that, I wonder, how did you test this GH Action if it works?
> 
> I only validated whether the YAML/syntax are correct. I'm not aware of any method to test a GH with shared secrets locally

GH Actions are difficult to test. I think what you need to do is to fork the repo and try the GH Actions in your fork.

File: .github/workflows/geth-with-protocol.yml
Comment (leszko) [inline .github/workflows/geth-with-protocol.yml:2]: One other thing you can consider is to add a parameter with `inputs`. Then, you could not hardcode `confluence` docker image tag, but the user could just input whatever tag they want.

Anyway, I leave it up to you.

**Review by leszko — APPROVED** [review]
Added one more minor comment. Other than that LGTM 👍 

One more note: Please test it in your fork before merging this PR to not break `master`.

File: .github/workflows/geth-with-protocol.yml
Comment (RiccardoBiosas) [reply .github/workflows/geth-with-protocol.yml:2]: IMHO I think it makes sense to agree to name it `confluence` since it seems all the protocol-related docker releases have been tied up to breaking changes in the protocol (i.e.: the previous one was tagged `streamflow`)

Anyway good thing you asked me to test the GH on a fork, I actually missed a -t flag in the docker build command

**Pull request merged**
